Contributing-code-to-Oppia.md

```diff
@@ -1,306 +1,211 @@
+8. If you are making a contribution which involves changing some user interface or introducing a new feature, it is good to start with a design doc to avoid wasting efforts later. Follow our [[guide for writing design docs|Writing-design-docs]].
 
-This will enable you to get reviews from your mentors and other Oppia contributors easily without needing to schedule a meeting, and make it easier to add additional reviewers later if needed. You can find more information on [this wiki page](https://github.com/oppia/oppia/wiki/Writing-design-docs) about how to write these docs, as well as best practices for responding to doc reviews.
+### Notes
 
-## Juggling multiple pull requests
+* Our central development branch is `develop`, which should be clean and ready for release at any time. All changes should be done in feature branches based off of `develop`.
 
-After you've submitted a PR, and are waiting for it to get reviewed, you might want to start working on a new issue in the meantime. Git allows you to do this seamlessly! Here are some tips that might be helpful:
+* Sometimes, GitHub comments in the main conversation thread don't have a reply box. This seems to be a quirk with GitHub: apparently the reply box doesn't show up on outdated threads if you're currently in the middle of a review (or a reply to someone else's review) that you haven't submitted. After you submit the review, the reply field should show up again.
 
-- Always prioritize following up on your existing PRs and getting them ready for the next round of review. In particular, only start working on new issues/PRs after **all** the other PRs you're working on are in the "waiting for review" stage. This helps avoids creating a logjam in the build queues.
+* If you want to do a codebase change that is large and somewhat repetitive, do a small trial PR first for a limited subset of the change, and check with reviewers whether the approach makes sense.  Then, only do the full change after getting that trial PR merged (or at least approved by all reviewers).
 
-- When starting work on a new issue, make sure you're on the `develop` branch first (and not a branch associated with any of your existing PRs) before checking out a new branch. In other words, run:
+  This helps because, if you just did the full PR at the outset instead, then if a reviewer requests changes to the approach, you'd need to go back and modify all the files. On the other hand, with a trial PR, addressing an initial round of changes is less work, and by the time you get to the full PR, you'd already know what you need to do!
 
-   git checkout develop
-   git checkout -b {{NEW_BRANCH_NAME}}
+* To find the author of a particular change in a file, run this command:
```

Comment:
Is it worth explaining how to do this in the GitHub UI too? It's kind of nice to see who last changed a specific line, so that might be even more useful.

Reply:
done